File: Invalidmove.py
import sys
import play1
import Board

def invalid():

    print("Invalid move\nat least one coin should flip")
    print("Enter 'R' to replay your turn")
    print("If no valid moves are available enter 'P' to pass your turn")

    while True:
        ch = input().strip().lower()
        if ch == "r":
            play1.j -= 1
            break
        elif ch == "p":
            play1.m += 1
            break
        else:
            print("Enter 'R' to replay, 'P' to pass", file=sys.stderr)

    i = play1.in_ // 10
    j = play1.in_ % 10
    play1.b1[i][j] = "_-_|"
    Board.display(play1.b1)

File: games_java/invalidmove.java
import java.io.*;
class invalidmove
{
    public static void invalid(play1 p)throws Exception
    {
        BufferedReader br=new BufferedReader(new InputStreamReader(System.in));
        System.out.println("Invalid move\nat least one coin should flip");
        System.out.println("Enter 'R' to replay your turn");
        System.out.println("If no valid moves are available enter 'P' to pass your turn");
        String ch= "";
        do
        {
            ch=br.readLine();
            ch=ch.trim();
            if(!ch.equalsIgnoreCase("R")&&!ch.equalsIgnoreCase("P"))
            {
                System.out.println("Enter 'R' to replay, 'P' to pass");
            }
        }
        while(!ch.equalsIgnoreCase("R")&&!ch.equalsIgnoreCase("P"));
        if(ch.equalsIgnoreCase("r"))
        {
            --p.j;
        }
        else if(ch.equalsIgnoreCase("p"))
        {
            ++p.m;
        }
        int i=p.in/10;
        int j=p.in%10;
        p.b1[i][j]="_-_|";
        Board.display(p.b1);
    }
}
           
        